src/lib.rs

```diff
@@ -6,3 +6,6 @@ pub mod operations;
 pub mod state;
 
 pub use crate::error::ContractError;
+
+#[cfg(test)]
+mod tests;
```

Comment:
why not have the UTs as mods living in the same files as the functions they test?